samples/snippets/fsharp/azure/file-storage.fsx

```diff
@@ -14,127 +17,139 @@ let storageConnString = "..." // fill this in from your storage account
 let storageConnString = 
     CloudConfigurationManager.GetSetting("StorageConnectionString")
 *)
-//
-// Parse the connection string.
-//
-
-// Parse the connection string and return a reference to the storage account.
-let storageAccount = CloudStorageAccount.Parse(storageConnString)
 
 //
 // Create the File service client.
 //
 
-let fileClient = storageAccount.CreateCloudFileClient()
+let share = ShareClient(storageConnString, "shareName")
 
 //
 // Create a file share.
 //
 
-let share = fileClient.GetShareReference("myfiles")
-share.CreateIfNotExists()
+share.CreateIfNotExistsAsync()
 
 //
-// Create a root directory and a subdirectory
+// Create a directory
 //
 
-let rootDir = share.GetRootDirectoryReference()
-let subDir = rootDir.GetDirectoryReference("myLogs")
-subDir.CreateIfNotExists()
+// Get a reference to the directory
+let directory = share.GetDirectoryClient("directoryName")
+
+// Create the directory if it doesn't already exist
+directory.CreateIfNotExistsAsync()
 
 //
-// Upload a file to a subdirectory
+// Upload a file to the sample directory
 //
 
-let file = subDir.GetFileReference("log.txt")
-file.UploadText("This is the content of the log file")
+let file = directory.GetFileClient("fileName")
+let stream = File.OpenRead("localFilePath")
+file.Create(stream.Length)
+file.UploadRange(
+    HttpRange(0L, stream.Length),
+    stream)
 
 //
-// Download a file to a local fie
+// Download a file to a local file
 //
 
-file.DownloadToFile("log.txt", FileMode.Append)
+let download = file.Download()
+let downStream = File.OpenWrite("Save_Download_Path")
+download.Value.Content.CopyTo(stream)
 
 //
 // Set the maximum size for a file share.
 //
 
 // stats.Usage is current usage in GB
-let stats = share.GetStats()
-share.FetchAttributes()
+let ONE_GIBIBYTE = 10737420000L // Number of bytes in 1 gibibyte
+let stats = share.GetStatistics().Value
+let currentGiB = (int)(stats.ShareUsageInBytes / ONE_GIBIBYTE)
```

Comment:
```suggestion
let currentGiB = int (stats.ShareUsageInBytes / ONE_GIBIBYTE)
```

This is more idiomatic F#, because `int` is actually a conversion _function_ instead of a name you cast to.